practice1.cpp

```diff
@@ -0,0 +1,178 @@
+#include <iostream>
+#include <fstream>
+using namespace std;
+
+
+/////////////////////////////////////////////////////////////////////
+struct Footbolist {
+    string surname = "";
+    string role = "";
+    short age = 0;
+    short games_count = 0;
+    int goal_count = 0;
+};
+/////////////////////////////////////////////////////////////////////
+void PrintFootbolist(Footbolist* footbolist) {
+    cout << "surname: " << footbolist->surname << " role: " << footbolist->role << 
+        " age: " << footbolist->age << " games count: " << footbolist->games_count << 
+        " goal count: " << footbolist->goal_count << endl;
+}
+/////////////////////////////////////////////////////////////////////
+void BestForward(Footbolist* footbolist, int n) {
+    Footbolist bestForward = footbolist[0];
+    for (int i = 0; i < n; i++) {
+        if (1.0f*(footbolist+i)->goal_count / (footbolist+i)->games_count >
+            1.0f*(&bestForward)->goal_count / (&bestForward)->games_count) bestForward = footbolist[i];
+    }
+    PrintFootbolist(&bestForward);
+}
+bool isLessThanFiveGames(Footbolist* footbolist) {
+    return footbolist->games_count < 5;
+}
+void LessThanFiveGames(Footbolist* footbolist, int n) {
+    for (int i = 0; i < n; i++) 
+        if (isLessThanFiveGames(footbolist + i)) 
+            PrintFootbolist(footbolist + i);
+}
+/////////////////////////////////////////////////////////////////////
+bool SameFootbolist(Footbolist footbolist1, Footbolist footbolist2) {
+    return (&footbolist1)->surname == (&footbolist2)->surname &&
+        (&footbolist1)->role == (&footbolist2)->role &&
+        (&footbolist1)->age == (&footbolist2)->age &&
+        (&footbolist1)->games_count == (&footbolist2)->games_count &&
+        (&footbolist1)->goal_count == (&footbolist2)->goal_count;
+}
+int FindFootbolist(Footbolist* footbolist, int n, Footbolist footbolist0) {
+    for (int i = 0; i < n; i++) if (SameFootbolist(footbolist[i], footbolist0)) return i;
+    return -1;
+}
+Footbolist SetFootbolist() {
+    Footbolist footbolist;
+    cout << "Footbolist's surname: ";
+    cin >> (&footbolist)->surname;
+    cout << "Footbolist's role: ";
+    cin >> (&footbolist)->role;

```

Comment:
Here you do not have any check if the role of the football player is correct. Do "role" an enum or do checker for role to be correct